cmd/common_test.go

```diff
@@ -165,3 +167,131 @@ func TestIsMaster(t *testing.T) {
 		assert.Equal(t, tc.isMaster, isMaster(), tc.name)
 	}
 }
+
+func TestMapToCISVersion(t *testing.T) {
+
+	viperWithData, err := loadConfigForTest()
+	if err != nil {
+		t.Fatalf("Unable to load config file %v", err)
+	}
+	kubeToCISMap, err := loadVersionMapping(viperWithData)
+	if err != nil {
+		t.Fatalf("Unable to load config file %v", err)
+	}
+
+	cases := []struct {
+		kubeVersion string
+		succeed     bool
+		exp         string
+	}{
+		{kubeVersion: "1.11", succeed: true, exp: "cis-1.3"},
+		{kubeVersion: "1.12", succeed: true, exp: "cis-1.3"},
+		{kubeVersion: "1.13", succeed: true, exp: "cis-1.4"},
+		{kubeVersion: "1.16", succeed: true, exp: "cis-1.4"},
+		{kubeVersion: "unknown", succeed: false, exp: ""},
+	}
+	for _, c := range cases {
+		rv := mapToCISVersion(kubeToCISMap, c.kubeVersion)
+		if c.exp != rv {
+			t.Errorf("mapToCISVersion kubeversion: %q Got %q expected %s", c.kubeVersion, rv, c.exp)
+		}
+	}
+}
+
+func TestLoadVersionMapping(t *testing.T) {
+	setDefault := func(v *viper.Viper, key string, value interface{}) *viper.Viper {
+		v.SetDefault(key, value)
+		return v
+	}
+
+	viperWithData, err := loadConfigForTest()
+	if err != nil {
+		t.Fatalf("Unable to load config file %v", err)
+	}
+
+	cases := []struct {
+		n       string
+		v       *viper.Viper
+		succeed bool
+	}{
+		{n: "empty", v: viper.New(), succeed: false},
+		{
+			n:       "novals",
+			v:       setDefault(viper.New(), versionMapping, "novals"),
+			succeed: false,
+		},
+		{
+			n:       "good",
+			v:       viperWithData,
+			succeed: true,
+		},
+	}
+	for _, c := range cases {
+		rv, err := loadVersionMapping(c.v)
+		if c.succeed {
+			if err != nil {
+				t.Errorf("[%q]-Unexpected error: %v", c.n, err)
+			}
+
+			if len(rv) == 0 {
+				t.Errorf("[%q]-missing mapping value", c.n)
+			}
+		} else {
+			if err == nil {
+				t.Errorf("[%q]-Expected error but got none", c.n)
+			}
+		}
+	}
+}
+
+func TestGetBenchmarkVersion(t *testing.T) {
+	viperWithData, err := loadConfigForTest()
+	if err != nil {
+		t.Fatalf("Unable to load config file %v", err)
+	}
+
+	cases := []struct {
+		n                string
+		kubeVersion      string
+		benchmarkVersion string
+		v                *viper.Viper
+		exp              string
+		succeed          bool
+	}{
+		{n: "both versions", kubeVersion: "1.11", benchmarkVersion: "cis-1.3", exp: "cis-1.3", v: viper.New(), succeed: false},
+		{n: "no version", kubeVersion: "", benchmarkVersion: "", v: viperWithData, exp: "cis-1.3", succeed: true},
```

Comment:
I'm not sure about this "no version" test. If the user doesn't specify either kubeVersion or benchmarkVersion, we should be using `getKubeVersion()` to find the running version of kubernetes, and failing if we can't get it. In unit testing we shouldn't be getting a running version of kubernetes, should we? (Unless that is mocked out and I am failing to see that)

Reply:
yeah that's a bug. I'll fix that.